### ISSUE TITLE
Code injection problem on Linux

### ISSUE THREAD

**Issue by rdbo** [rdbo/libmem#4]
I found out that external functions related to code injection on Linux might cause the target program to crash.
The functions doing code injection are:
```
mem_ex_protect
mem_ex_allocate
mem_ex_deallocate
mem_ex_load_library
```

**Comment by rdbo** [comment]
Mostly fixed, there are still 2 issues though:
1. For some reason, every 32bit syscall that I tried worked, _except_ mmap, which is used to allocate memory in mem_ex_allocate.
2. As mem_ex_allocate is not working properly in 32bit right now, mem_ex_load_library won't work in 32bit either, because it needs to allocate memory in the target process to inject the dlopen call and the library path into its virtual memory.

This issues need deeper looking, closing this for now.

**Comment by rdbo** [comment]
Fixed mem_ex_allocate for 32 bits. Now all the syscalls works, but I still have to work on the 32 bit version of mem_ex_load_library.

**Comment by rdbo** [comment]
Fixed mem_ex_load_library for x64. Now I have to see why is it not working on x86. This issue will definitely be fixed on 2.12

**Comment by rdbo** [comment]
mem_ex_load_library is now fixed for 32 bits. Finally.
